tests/integration/defs/stress_test/stress_test.py

```diff
@@ -0,0 +1,941 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Stress test script for inference of model using TensorRT-LLM with PyTorch/TRT backend.
+This script is used for stress testing inference performance using trtllm-serve and genai-perf.
+"""
+import contextlib
+import json
+import os
+import re
+import subprocess
+import threading
+import time
+from dataclasses import dataclass, field
+from glob import glob
+from typing import List, Optional, Tuple
+
+import pandas as pd
+import pytest
+import requests
+from defs.conftest import get_device_count, get_device_memory, llm_models_root
+from defs.trt_test_alternative import (Popen, altered_env, cleanup_process_tree,
+                                       print_info, print_warning)
+
+# Define a constant for process termination timeouts
+GRACEFUL_TERMINATION_TIMEOUT = 4  # seconds
+
+
+@dataclass(frozen=True)
+class ServerConfig:
+    """Dataclass to store server configuration for trtllm-serve"""
+    port: int = 8000
+    host: str = "0.0.0.0"
+    wait_interval: int = 10  # seconds
+    max_wait_seconds: int = 360  # 6 mins <- Larger model need longer wait time
+    health_check_timeout: float = 8  # seconds <- Make it smaller than wait_interval
```

Comment:
8 seconds seems very optimistic. Considering load models from NFS, can be really slow in the first time. 

Maybe not set this, have a env to control this?

Reply:
This is purely periodically check if serve is ready and could let us know exactly how long time model will be ready after loading from NFS, etc. And can be changed ad-hoc after pre-run.